cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -93,9 +98,11 @@ protected AbstractContext(Program program, ProgramOptions programOptions,
                             Set<String> datasets, DatasetFramework dsFramework, TransactionSystemClient txClient,
                             DiscoveryServiceClient discoveryServiceClient, boolean multiThreaded,
                             @Nullable MetricsCollectionService metricsService, Map<String, String> metricsTags,
-                            SecureStore secureStore, SecureStoreManager secureStoreManager) {
+                            SecureStore secureStore, SecureStoreManager secureStoreManager,
+                            @Nullable PreviewId previewId, DebugLoggerFactory debugLoggerFactory) {
     this(program, programOptions, datasets, dsFramework, txClient,
-         discoveryServiceClient, multiThreaded, metricsService, metricsTags, secureStore, secureStoreManager, null);
+         discoveryServiceClient, multiThreaded, metricsService, metricsTags, secureStore, secureStoreManager, null,
+         previewId, debugLoggerFactory);
```

Comment:
Is it? I thought under preview mode, the preview id is generated and is used as the application id? @sagarkapare can you clarify?